client/src/client.rs

```diff
@@ -776,7 +779,24 @@ impl<B, E, Block, RA> Client<B, E, Block, RA> where
 				is_new_best,
 				storage_changes,
 				retracted,
-			})
+			});
+		} else {
+			match origin {
+				// send a notification for the offchain workers
+				// which would not be notifice in the network initial sync
+				// case otherwise (see previous origin match statement)
+				BlockOrigin::NetworkInitialSync => {
+					self.notify_initial_sync_imported(ImportSummary {
```

Comment:
I think that `all_blocks` stream should not contain all this data, but rather since it's on a performance-critical path it should be just minimum necessary (so no storage changes, retracted blocks, etc).

Reply:
I solved this for now with a new `ChainEvent` variant (`ChainEvent::AnyNewBlock`) being that entity.